packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1523,7 +1533,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     if (props.instanceMonitoring) {
       throw new Error('Setting \'instanceMonitoring\' must not be set when \'launchTemplate\' or \'mixedInstancesPolicy\' is set');
     }
-    if (props.associatePublicIpAddress) {
+    if (props.associatePublicIpAddress || props.associatePublicIpAddress === false) {
```

Comment:
I'm not sure I get why setting it to false should be an issue here. It seems like we're mishandling it somewhere else and that's where we need to apply the fix.

Reply:
This could probably be simplified to `if (props.associatePublicIpAddress !== undefined)`.

It looks like if you specify the `launchTemplate` then you must specify the `associatePublicIpAddress` on the `LaunchTemplate` and not the `AutoScalingGroup`. The problem in the linked issue is that `associatePublicIpAddress` must be specified in the `LaunchTemplate.networkInterfaces` property which is not currently supported, which means that you can't use `launchTemplate` and set `associatePublicIpAddress: false` currently. Which is not great.

I think long term we should deprecate the `AutoScalingGroup` construct and create a `AutoScalingGroupV2` (or something) which accepts an `ILaunchConfiguration` instead of just having all the properties for a `LaunchConfiguration` in the `AutoScalingGroupProps`.

Reply:
Yep, the design here isn't too great. However I'm not sure what we can do for now except ensure the error gets thrown.